application/common/models/User.php

```diff
@@ -199,6 +212,73 @@ public function updateProfileIfNeeded($firstName, $lastName, $username, $email)
     }
 
     /**
+     * In case where email address has been reassigned to $this user, refresh profile for user
+     * with email address currently in case their email address has changed too
+     * @param string $email
+     * @return bool
+     * @throws \Exception
+     */
+    public static function refreshPersonnelDataForUserWithSpecificEmail($email)
```

Comment:
Should we consider other places in the db, e.g., methods, where we should wipe the current associated data?